### PR TITLE
Tree-shake ready

### PR DESCRIPTION
**Pull request by bpierre**
This optimization will make it possible to include components like [Markdown](https://github.com/aragon/aragon-ui/pull/395) without affecting the performance of apps that are not using it.

Some numbers:

| App                       | Before    | After     | Decrease           | Notes                                                                                                                                                                                              |
| ------------------------- | --------- | --------- | ------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Basic app (Main + Button) | 144.21 KB | 76.99 KB  | 46.6%              |                                                                                                                                                                                                    |
| aragon/aragon (Client)    | 1040 KB   | 906.79 KB | 12.8% 🧐           |                                                                                                                                                                                                    |
| Finance                   | 390 KB    | 307 KB    | 21.3% :open_mouth: |                                                                                                                                                                                                    |
| Tokens                    | 348.15 KB | 196.73 KB | 43.5% :scream:     | Also removed [web3-utils](https://bundlephobia.com/result?p=web3-utils@1.2.6), which was breaking with Parcel’s tree shaking. [See pull request](https://github.com/aragon/aragon-apps/pull/1076). |

### CODE REVIEW

File: src/components/index.js [hunk 1]
```diff
@@ -1,17 +1,10 @@
...
-export { DataView } from './DataView/DataView'
-export { DiscButton } from './DiscButton/DiscButton'
-export { Info } from './Info'
-export { Layout, useLayout } from './Layout/Layout'
-export { Link, ExternalLink, SafeLink } from './Link'
-export { Pagination } from './Pagination/Pagination'
-export { SearchInput } from './Input/SearchInput'
-export { Tag } from './Tag/Tag'
+export { ExternalLink, SafeLink } from './Link/LinkDeprecated'
 export { ToastHub, Toast, useToast } from './ToastHub/ToastHub'
```
Comment (Evalir) [inline src/components/index.js]: Should we move this down to the corresponding part of the file?

File: src/theme-legacy/index.js [hunk 2]
```diff
@@ -1 +1 @@
-export { theme as default, theme, themeDark, brand, colors } from './palettes'
+export { theme, themeDark, brand, colors } from './palettes'
```
Comment (sohkai) [inline src/theme-legacy/index.js]: Is this a breaking change?

File: rollup.config.js [hunk 3]
```diff
@@ -1,62 +1,81 @@
...
+    input: {
+      index: 'src/index.js',
+      components: 'src/components/index.js',
+      hooks: 'src/hooks/index.js',
+      providers: 'src/providers/index.js',
+      icons: 'src/icons/index.js',
+      themeLegacy: 'src/theme-legacy/index.js',
+      theme: 'src/theme/index.js',
+      utils: 'src/utils/index.js',
+      style: 'src/style/index.js',
```
Comment (sohkai) [inline rollup.config.js]: Wondering if we should also use `glob` to make this list; do you think we'll easily forget to update it in the future?

File: src/components/index.js (hunk 1 above)
Comment (bpierre) [reply src/components/index.js]: Done in https://github.com/aragon/aragon-ui/pull/709/commits/1aa7a5795ee3e8faa9fdfd0e3c2688f6d00e9b8e and https://github.com/aragon/aragon-ui/pull/709/commits/cdecbed342e623559a5fc762d3e20c4020bbd241

File: src/theme-legacy/index.js (hunk 2 above)
Comment (bpierre) [reply src/theme-legacy/index.js]: No it was only for internal imports, `default` was never exported: https://github.com/aragon/aragon-ui/blob/master/src/index.js#L5

File: rollup.config.js (hunk 3 above)
Comment (bpierre) [reply rollup.config.js]: Yes I think we can do this. We’ll have to be careful if one day we add a private `dir/index.js` here, but we are not doing it right now.

File: rollup.config.js [hunk 4]
```diff
@@ -50,10 +60,26 @@ export default (async () => {
...
+        exports: 'named',
       },
     ],
     external: ['react', 'react-dom', 'styled-components'],
-    treeshake: true,
+    treeshake: production
+      ? {
+          moduleSideEffects: false,
+
+          // Reminder: getters are going to be used to deprecate theme-legacy, so
```
Comment (sohkai) [inline rollup.config.js]: Maybe we can add a TODO or a companion comment in `theme.js` to remember to do this?

**Pull request merged**